static/js/unknown-thing.js

```diff
@@ -0,0 +1,354 @@
+/**
+ * Unknown thing.
+ *
+ * UI element representing an unknown thing type.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+'use strict';
+
+/* globals Thing, ThingDetailLayout */
+
+/**
+ * UnknownThing Constructor (extends Thing).
+ *
+ * @param Object description Thing description object.
+ */
+var UnknownThing = function(description, format) {
+  this.displayedProperties = {};
+  if (description.properties) {
```

Comment:
This should also include if (format === 'htmlDetail') or the updateProperty method later on might call update() on an unattached detail

Reply:
All of the `update()` methods already handle that case.

Reply:
Doing that also ends up breaking rendering of the HTML views, as the code has to look up if there are additional properties or not to justify a full expanded view.

Reply:
Ah, sounds good. Thanks for the explanation!